op-conductor/health/monitor.go

```diff
@@ -0,0 +1,143 @@
+		interval:       interval,
+		healthUpdateCh: make(chan bool),
+		rollupCfg:      rollupCfg,
+		safeInterval:   safeInterval,
+		minPeerCount:   minPeerCount,
+		node:           node,
+		p2p:            p2p,
+	}
+}
+
+// SequencerHealthMonitor monitors sequencer health.
+type SequencerHealthMonitor struct {
+	log  log.Logger
+	done chan struct{}
+	wg   sync.WaitGroup
+
+	rollupCfg           *rollup.Config
+	safeInterval        uint64
+	minPeerCount        uint64
+	interval            uint64
+	healthUpdateCh      chan bool
+	lastSeenUnsafeBlock eth.L2BlockRef
+	lastSeenSafeBlock   eth.L2BlockRef
+
+	node dial.RollupClientInterface
+	p2p  p2p.API
+}
+
+var _ HealthMonitor = (*SequencerHealthMonitor)(nil)
+
+// Start implements HealthMonitor.
+func (hm *SequencerHealthMonitor) Start() error {
+	hm.log.Info("starting health monitor")
+	hm.wg.Add(1)
+	go hm.loop()
+
+	hm.log.Info("health monitor started")
+	return nil
+}
+
+// Stop implements HealthMonitor.
+func (hm *SequencerHealthMonitor) Stop() error {
+	hm.log.Info("stopping health monitor")
+	close(hm.done)
+	hm.wg.Wait()
+
+	hm.log.Info("health monitor stopped")
+	return nil
+}
+
+// Subscribe implements HealthMonitor.
+func (hm *SequencerHealthMonitor) Subscribe() <-chan bool {
+	return hm.healthUpdateCh
+}
+
+func (hm *SequencerHealthMonitor) loop() {
+	defer hm.wg.Done()
+
+	duration := time.Duration(hm.interval) * time.Second
+	ticker := time.NewTicker(duration)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-hm.done:
+			return
+		case <-ticker.C:
+			hm.healthUpdateCh <- hm.healthCheck()
+		}
+	}
+}
+
+// healthCheck checks the health of the sequencer by 3 criteria:
+// 1. unsafe head is progressing per block time
+// 2. safe head is progressing every configured batch submission interval
+// 3. peer count is above the configured minimum
+func (hm *SequencerHealthMonitor) healthCheck() bool {
+	ctx := context.Background()
+	status, err := hm.node.SyncStatus(ctx)
+	if err != nil {
+		hm.log.Error("health monitor failed to get sync status", "err", err)
+		return false
+	}
+	hm.lastSeenUnsafeBlock = status.UnsafeL2
+	hm.lastSeenSafeBlock = status.SafeL2
+
+	now := uint64(time.Now().Unix())
+	// allow at most one block drift for unsafe head
+	if now-hm.lastSeenUnsafeBlock.Time > hm.interval+hm.rollupCfg.BlockTime {
+		hm.log.Error("unsafe head is not progressing", "lastSeenUnsafeBlock", hm.lastSeenUnsafeBlock)
+		return false
+	}
+
+	if now-hm.lastSeenSafeBlock.Time > hm.safeInterval {
+		hm.log.Error("safe head is not progressing", "safe_head_time", hm.lastSeenSafeBlock.Time, "now", now)
+		return false
+	}
+
+	stats, err := hm.p2p.PeerStats(ctx)
+	if err != nil {
+		hm.log.Error("health monitor failed to get peer stats", "err", err)
+		return false
+	}
+	if uint64(stats.Connected) < hm.minPeerCount {
+		hm.log.Error("peer count is below minimum", "connected", stats.Connected, "minPeerCount", hm.minPeerCount)
+		return false
+	}
+
+	return true
+}
```

Comment:
Unless I'm misunderstanding, the `lastSeenUnsafeBlock` and `lastSeenSafeBlock` are reset each loop iteration, so I think we can make them local instead of `SequencerHealthMonitor` state variables. Something like:

```suggestion
func (hm *SequencerHealthMonitor) healthCheck() bool {
	ctx := context.Background()
	status, err := hm.node.SyncStatus(ctx)
	if err != nil {
		hm.log.Error("health monitor failed to get sync status", "err", err)
		return false
	}
	lastSeenUnsafeBlock := status.UnsafeL2
	lastSeenSafeBlock := status.SafeL2

	now := uint64(time.Now().Unix())
	// allow at most one block drift for unsafe head
	if now - lastSeenUnsafeBlock.Time > hm.interval + hm.rollupCfg.BlockTime {
		hm.log.Error("unsafe head is not progressing", "lastSeenUnsafeBlock", lastSeenUnsafeBlock)
		return false
	}

	if now - lastSeenSafeBlock.Time > hm.safeInterval {
		hm.log.Error("safe head is not progressing", "safe_head_time", lastSeenSafeBlock.Time, "now", now)
		return false
	}

	stats, err := hm.p2p.PeerStats(ctx)
	if err != nil {
		hm.log.Error("health monitor failed to get peer stats", "err", err)
		return false
	}
	if uint64(stats.Connected) < hm.minPeerCount {
		hm.log.Error("peer count is below minimum", "connected", stats.Connected, "minPeerCount", hm.minPeerCount)
		return false
	}

	return true
}
```

Reply:
Good call, removed them entirely.